keras/engine/topology.py

```diff
@@ -2469,14 +2469,27 @@ def save_weights_to_hdf5_group(self, f):
                 else:
                     param_dset[:] = val
 
-    def load_weights(self, filepath):
+    def load_weights(self, filepath, by_name=False):
         '''Load all layer weights from a HDF5 save file.
+
+        If `by_name` is False (default) weights are loaded
+        in flattened array order, meaning the architecture
```

Comment:
"weights are loaded based on the network's topology" might be a clearer way to put it.